src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java

```diff
@@ -144,4 +155,57 @@ public FileDownloadController(ExperimentFileLocationService experimentFileLocati
             zipOutputStream.close();
         }
     }
+
+    @GetMapping(value = "json/experiments/download/zip/check",
+            produces = "application/json;charset=UTF-8")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public String
+    checkMultipleExperimentsFileValid(@RequestParam(value = "accession", defaultValue = "") List<String> accessions)
+    {
+        List<Experiment> experiments = new ArrayList<>();
+        for (var accession : accessions) {
+            try {
+                experiments.add(experimentTrader.getPublicExperiment(accession));
+            } catch (Exception e) {
+                LOGGER.debug("Invalid experiment accession: {}", accession);
+            }
+        }
+        ImmutableList<ExperimentFileType> fileTypeCheckList = ImmutableList.of(
+                ExperimentFileType.QUANTIFICATION_RAW,
+                ExperimentFileType.NORMALISED,
+                ExperimentFileType.EXPERIMENT_DESIGN);
+
+       Map<String, List<String>> inValidFilesList = new HashMap<>();
+
+        if (!experiments.isEmpty()) {
+            for (var experiment : experiments) {
```

Comment:
You can change this double `for` loop to use streams instead. It’s an interesting exercise that I encourage you to try and will let you get quite deep with the streams API.

As a hint, build `invalidFilesList` first with one entry per experiment, but only the accessions with missing files will have a non-empty list as a value, and afterwards you can filter out those map entries.